driver-core/src/test/functional/com/mongodb/operation/FindAndDeleteOperationSpecification.groovy

```diff
@@ -107,4 +160,113 @@ class FindAndDeleteOperationSpecification extends OperationFunctionalSpecificati
         returnedDocument == pete
     }
 
+    def 'should create the expected command'() {
+        given:
+        def cannedResult = new BsonDocument('value', new BsonDocumentWrapper(BsonDocument.parse('{}'), new BsonDocumentCodec()))
+        def filter = BsonDocument.parse('{ filter : 1}')
+        def sort = BsonDocument.parse('{ sort : 1}')
+        def projection = BsonDocument.parse('{ projection : 1}')
+
+        def connection = Mock(Connection) {
+            _ * getDescription() >> Stub(ConnectionDescription) {
+                getServerVersion() >> serverVersion
+            }
+        }
+        def connectionSource = Stub(ConnectionSource) {
+            getConnection() >> connection
+        }
+
+        def writeBinding = Stub(WriteBinding) {
+            getWriteConnectionSource() >> connectionSource
+        }
+        FindAndDeleteOperation<Document> operation = new FindAndDeleteOperation<Document>(getNamespace(), documentCodec)
+        def expectedCommand = new BsonDocument('findandmodify', new BsonString(getNamespace().getCollectionName()))
+                .append('remove', BsonBoolean.TRUE)
+        when:
+        operation.execute(writeBinding)
+
+        then:
+        1 * connection.command(getNamespace().getDatabaseName(), expectedCommand, _, _, _) >> cannedResult
+        2 * connection.release()
+
+        when:
+        operation.filter(filter)
+                .sort(sort)
+                .projection(projection)
+                .maxTime(10, TimeUnit.MILLISECONDS)
+                .writeConcern(WriteConcern.W1)
+
+        expectedCommand.append('query', filter)
+                .append('sort', sort)
+                .append('fields', projection)
+                .append('maxTimeMS', new BsonInt64(10))
+
+        if (serverVersion.compareTo(new ServerVersion([3, 2, 0])) >= 0) {
+            expectedCommand.put('writeConcern', WriteConcern.W1.asDocument())
+        }
+
+        operation.execute(writeBinding)
+
+        then:
+        1 * connection.command(getNamespace().getDatabaseName(), expectedCommand, _, _, _) >> cannedResult
+        2 * connection.release()
+
+        where:
+        serverVersion << [new ServerVersion([3, 0, 0]), new ServerVersion([3, 2, 0])]
```

Comment:
Given above comment, should test with three different write concerns:

* ACKNOWLEDGED
* UNACKNOWLEDGED
* W1